PLT-374 - Jira create and destroy environment ticket actions

jake-naughton opened the pull request:
- These actions will be used when provisioning and destroying environments.
- Have tested these with successfully
- https://github.com/cartoncloud/infrastructure-tf/actions/runs/6753098658
- https://github.com/cartoncloud/infrastructure-tf/actions/runs/6753104261

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Adding Jira create ticket action

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Adding Jira create ticket action to local build

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Fixing build

jake-naughton committed:
Compile actions

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Fixing package json

jake-naughton committed:
Merge remote-tracking branch 'origin/tech/PLT-374' into tech/PLT-374

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Generating package lock

jake-naughton committed:
Compile actions

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Fixing input variable

jake-naughton committed:
Merge remote-tracking branch 'origin/tech/PLT-374' into tech/PLT-374

jake-naughton committed:
Compile actions

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Adding further debugging

jake-naughton committed:
Compile actions

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- adding summary

jake-naughton committed:
Merge remote-tracking branch 'origin/tech/PLT-374' into tech/PLT-374

jake-naughton committed:
Compile actions

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Adding delete action

jake-naughton committed:
Merge remote-tracking branch 'origin/tech/PLT-374' into tech/PLT-374

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Adding delete action to build workflow

jake-naughton committed:
Compile actions

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Fixing JQL

jake-naughton committed:
Merge remote-tracking branch 'origin/tech/PLT-374' into tech/PLT-374

jake-naughton committed:
Compile actions

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Adding delete API call

jake-naughton committed:
Merge remote-tracking branch 'origin/tech/PLT-374' into tech/PLT-374

jake-naughton committed:
Compile actions

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Adding further infor logging

jake-naughton committed:
Merge remote-tracking branch 'origin/tech/PLT-374' into tech/PLT-374

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Adding further info logging

jake-naughton committed:
Compile actions

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Fixing README

jake-naughton committed:
Merge remote-tracking branch 'origin/tech/PLT-374' into tech/PLT-374

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Fixing issue link output

jake-naughton committed:
Compile actions

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Fixing Readme

jake-naughton committed:
Merge remote-tracking branch 'origin/tech/PLT-374' into tech/PLT-374

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Fixing Readme

aarontwf commented on jira-environment-ticket-create/src/index.ts:
```diff
@@ -0,0 +1,56 @@
+    if (!createResponse.ok) {
+      core.error(`response code: ${createResponse.status}`);
+      core.error('response: ' + JSON.stringify(createResponseJson));
+      core.setFailed(`Failed to create environment ticket.`);
```
Does `setFailed` stop the execution here or should the following be wrapped in an `else`?

aarontwf commented on jira-environment-ticket-delete/src/index.ts:
```diff
@@ -0,0 +1,62 @@
+    if (!deleteResponse.ok) {
+      core.error(`response code: ${deleteResponse.status}`);
+      core.error('response: ' + JSON.stringify(deleteResponse.json()));
+      core.setFailed(`Failed to delete environment ticket.`);
```
Does `setFailed` stop the execution here or should the following be wrapped in an `else`?

jake-naughton replied:
Seems in other places we have just used:
`return;` 
To halt the execution of the script. I'll just use that.

jake-naughton committed:
PLT-374 - Jira create and destroy environment ticket actions
- Code Review: adding return to prevent script executing further after failing.

jake-naughton committed:
Compile actions
